packages/react/src/experimental/OneActionBar/index.tsx

```diff
@@ -139,49 +204,34 @@ export const OneActionBar = ({
               {warningMessage ? (
                 <Alert message={warningMessage} />
               ) : (
-                <>
+                <Fragment key="mobile-actions">
                   <MobileDropdown items={secondaryActions} />
-                  {primaryActions.length > 1 ? (
+                  {!singlePrimaryAction ? (
                     <OneDropdownButton
-                      items={primaryActions.map((action) =>
-                        isActionType(action)
-                          ? {
-                              value: action.label,
-                              label: action.label,
-                              icon: action.icon,
-                              critical: action.critical,
-                              description: action.description,
-                            }
-                          : { type: "separator" as const }
-                      )}
+                      items={primaryActionsDropdownItems}
                       onClick={(value) => {
-                        const action = primaryActions.find(
-                          (a) => isActionType(a) && a.label === value
-                        )
-                        if (action && isActionType(action)) {
-                          action.onClick?.()
-                        }
+                        const action = getActionByValue(value)
+                        ;(action as ActionType)?.onClick?.()
```

Comment:
something is wrong here. why do we have semicolon at the beginning?

Reply:
semicolon is needed as the line starts with a (, if no semicolon parser will interpret this is code for the previous line 
https://standardjs.com/rules#semicolons